x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -243,11 +255,39 @@ static void getMemoryAndProcessors(
                 if (Priority.LOW.equals(modelAssignment.getValue().getTaskParams().getPriority()) == false) {
                     // as assignments can be placed on different nodes, we only need numberOfThreadsPerAllocation here
                     extraSingleNodeProcessors = Math.max(extraSingleNodeProcessors, numberOfThreadsPerAllocation);
-                    extraProcessors += numberOfAllocations * numberOfThreadsPerAllocation;
+                    extraProcessors += numberOfRequestedAllocations * numberOfThreadsPerAllocation;
                 }
             } else if (assignment.getNodeRoutingTable().values().stream().allMatch(r -> r.getState().consumesMemory() == false)) {
                 // Ignore states that don't consume memory, for example all allocations are failed
                 continue;
+            } else if (AssignmentState.STARTED.equals(assignment.getAssignmentState()) && numMissingAllocations > 0) {
+                // the assignment has probably been updated, and we are waiting for the new allocations to be started
+                // we may need to scale up to accommodate the new allocations
+                if (numberOfAvailableProcessors(autoscalingContext, allocatedProcessorsScale) < numProcessorsNeeded) {
+                    // we don't have enough processors to start the new allocations, we need to scale up
+                    extraProcessors += numMissingAllocations * numberOfThreadsPerAllocation;
+                    final int extraProcessorsNeededForAssignment = numMissingAllocations * numberOfThreadsPerAllocation;
+                    logger.warn(
+                        () -> format(
+                            "trained model [%s] assigned to [%s], waiting for [%d] allocations to start; waiting on [%d] extra processors "
+                                + "to become available",
+                            modelAssignment.getKey(),
+                            Strings.arrayToCommaDelimitedString(modelAssignment.getValue().getStartedNodes()),
+                            numMissingAllocations,
+                            extraProcessorsNeededForAssignment
+                        )
+                    );
+                } else {
+                    logger.info(
+                        () -> format(
+                            "trained model [%s] assigned to [%s], waiting for [%d] allocations to start",
+                            modelAssignment.getKey(),
+                            Strings.arrayToCommaDelimitedString(modelAssignment.getValue().getStartedNodes()),
+                            numMissingAllocations
+                        )
+                    );
```

Comment:
I think this case could theoretically happen if the nodes were added by the allocations were not yet assigned